### PR TITLE
Refector baseCid in viewer into cid-impl

### PR DESCRIPTION
**Pull request by muxin**
Part of #6159

### CODE REVIEW

File: test/functional/test-viewer.js [hunk 1]
```diff
@@ -387,77 +386,6 @@ describe('Viewer', () => {
...
-      });
-      return p;
-    });
-
-    it('should not request cid for untrusted viewer', () => {
-      trustedViewer = false;
-      return expect(viewer.baseCid()).to.eventually.be.undefined;
-    });
-
-    it('should convert CID returned by legacy API to new format', () => {
```
Comment (lannka) [inline test/functional/test-viewer.js]: can we also move this test to the new place?

Comment (muxin) [inline test/functional/test-viewer.js]: moved

**Comment by muxin** [comment]
@lannka PTAL

File: test/functional/test-viewer.js [hunk 2]
```diff
@@ -387,77 +386,6 @@ describe('Viewer', () => {
...
-      return expect(viewer.baseCid()).to.eventually.equal(cidData);
-    });
-
-    it('should send message to store cid', () => {
-      const newCidData = JSON.stringify({time: 101, cid: 'cid-456'});
-      return expect(viewer.baseCid(newCidData))
-          .to.eventually.equal(newCidData);
-    });
-
-    it('should time out', () => {
```
Comment (lannka) [inline test/functional/test-viewer.js]: ah, forgot about this one :-)

**Comment by muxin** [comment]
@lannka PTAL

**Pull request merged**